### DISCUSSION TITLE
Allow overlapping for minimal window borders

### DISCUSSION THREAD

**Discussion by ArkadyBuryakov** [kovidgoyal/kitty#7956] [Ideas]
Right now with `draw_minimal_borders yes` we can draw borders between terminal space that looks very compact.
However it takes a little bit extra space and looks odd on with a thin border width.

It would be cool to take this option even further and allow borders to overlap, causing even thinner sleek-looking borders.
It could be either:
- new value for `draw_minimal_borders` = `overlap`
- new config option `overlap_minimal_borders yes`

Color for the border should be selected in following priority order:
1. Active window
2. Bell window
3. Inactive window

Here is a diagram that explains what i mean:
![image](https://github.com/user-attachments/assets/0a8ffa8d-18d5-463a-8364-27a9dbde69bf)

**Comment by kovidgoyal** [comment]
In principle, sure, but I dont have the time for this, PR is welcome.
